commit 0b438cb17ce1b92e5d2988e58897f0108bdd35e6
Author: Jocelyn Le Sage <jocelyn@le-sage.com>
Date:   Sun Feb 16 11:41:21 2025 -0500

    Make sure to set the owner of input and message box windows.
    This fixes issues seen in some environments (e.g. Linux with Openbox),
    where the state of the main window was changed when displaying an input
    or message box. The focus was also not set to the new displayed window.

diff --git a/VDF.GUI/Views/InputBoxView.xaml.cs b/VDF.GUI/Views/InputBoxView.xaml.cs
index e14a7c5..1b1b79a 100644
--- a/VDF.GUI/Views/InputBoxView.xaml.cs
+++ b/VDF.GUI/Views/InputBoxView.xaml.cs
@@ -52,6 +52,7 @@ namespace VDF.GUI.Views {
 			((InputBoxVM)DataContext).HasYesButton = (buttons & MessageBoxButtons.Yes) != 0;
 
 			InitializeComponent();
+			Owner = ApplicationHelpers.MainWindow;
 
 			if (!VDF.GUI.Data.SettingsFile.Instance.DarkMode)
 				RequestedThemeVariant = Avalonia.Styling.ThemeVariant.Light;
diff --git a/VDF.GUI/Views/MessageBoxView.xaml.cs b/VDF.GUI/Views/MessageBoxView.xaml.cs
index f02098b..55f0bca 100644
--- a/VDF.GUI/Views/MessageBoxView.xaml.cs
+++ b/VDF.GUI/Views/MessageBoxView.xaml.cs
@@ -57,6 +57,7 @@ namespace VDF.GUI.Views {
 
 			DataContext = vm;
 			InitializeComponent();
+			Owner = ApplicationHelpers.MainWindow;
 			if (!VDF.GUI.Data.SettingsFile.Instance.DarkMode)
 				RequestedThemeVariant = Avalonia.Styling.ThemeVariant.Light;
 			//Opened += MessageBoxView_Opened;
